tracing-error/src/lib.rs

```diff
@@ -1,79 +1,16 @@
-mod ctx;
-pub mod fmt;
+mod backtrace;
 mod layer;
-use std::error::Error;
-use tracing::{dispatcher, Metadata};
 
-pub use self::ctx::*;
+pub use self::backtrace::SpanBacktrace;
 pub use self::layer::ErrorLayer;
 
-pub struct ContextError {
-    inner: Box<dyn Error + Send + Sync>,
-    context: Context,
-}
-
-impl ContextError {
-    pub fn from_error(error: Box<dyn Error + Send + Sync + 'static>) -> Self {
-        ContextError {
-            inner: error,
-            context: Context::current(),
+#[macro_export]
+macro_rules! try_bool {
+    ($e:expr, $dest:ident) => {{
+        let ret = $e.unwrap_or_else(|e| $dest = Some(e));
+        if $dest.is_some() {
+            return false;
         }
-    }
-
-    pub fn context(&self) -> &Context {
-        &self.context
-    }
-
-    pub fn span_backtrace(&self) -> fmt::SpanBacktrace<&Self> {
-        fmt::SpanBacktrace::new(self)
-    }
-}
-
-pub trait TraceError: Error {
-    fn in_context(self) -> ContextError
-    where
-        Self: Sized + Send + Sync + 'static,
-    {
-        ContextError::from_error(Box::new(self))
-    }
-
-    fn context(&self) -> Option<&Context>
-    where
-        Self: Sized + 'static,
-    {
-        let cx = (self as &dyn Error)
-            .downcast_ref::<ContextError>()?
-            .context();
-        Some(cx)
-    }
-
-    fn span_backtrace(&self) -> fmt::SpanBacktrace<&(dyn Error + 'static)>
-    where
-        Self: Sized + 'static,
-    {
-        fmt::SpanBacktrace::new(self as &dyn Error)
-    }
-}
```

Comment:
I think we're going to want to bring something like this back before releasing this crate, but I'm fine with deleting it for now if you want to redo the API...

Reply:
Yea, I'd prefer to try implementing something local in one of my crates to experiment with this API rather than commit to something up front where I don't 100% understand how its intended to be used.